### DISCUSSION TITLE
The request URL with the port, But response the next url without the port

### DISCUSSION THREAD

**Discussion by rikeitou-0191632** [encode/django-rest-framework#8685] [General] [answered]
<pre>
Request URL: http://xxx.xxx.xxx.xxx:81/aaa
</pre>
<pre>
Response content: 
{
  "next": "http://xxx.xxx.xxx.xxx/aaa",
  "previous": "http://xxx.xxx.xxx.xxx/aaa",
  "current": "http://xxx.xxx.xxx.xxx/aaa"
}
</pre>

Nginx
<pre>
upstream django_server {
    server django:8001;
}

server {
    listen       81;
    location /aaa {
         proxy_pass http://django_server;
         ...
    }
</pre>

**Comment by lovelydinosaur** [comment, marked as answer]
To debug this you'd need to dig into Django's `request.build_absolute_uri()` and figure out what's going wrong. Possibly some misconfiguration.

I'd start by trying some debugging like...

```python
print(request.build_absolute_uri("/redirect"))   # Check if build_redirect_uri is including the port or not.
print(request.get_host())  # Show what host:port is being used for the absolute URIs.
print(request.META)  # Show the raw WSGI input to the request, which could help us dig out any misconfiguration issues.
```

Relevant bits of Django source code...

* [`Http.build_absolute_uri()`](https://github.com/django/django/blob/fa9ac16c1345a7fb6ad500c84961a954529ba2b9/django/http/request.py#L214)
* [`Http.get_host()`](https://github.com/django/django/blob/fa9ac16c1345a7fb6ad500c84961a954529ba2b9/django/http/request.py#L143)

**Reply by rikeitou-0191632** [reply]
Thank you very much.

The setting of Nginx is wrong.
<pre>
upstream django_server {
    server django:8001;
}

server {
    listen       81;
    location /aaa {
         proxy_pass http://django_server;
         proxy_set_header Host $http_host;  ← modified here. $host to $http_host
         ...
    }
</pre>
